converter/internal/otelcolconvert/converter_otlphttpexporter.go

```diff
@@ -0,0 +1,94 @@
+package otelcolconvert
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/alecthomas/units"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/exporter/otlphttp"
+	"github.com/grafana/agent/converter/diag"
+	"github.com/grafana/agent/converter/internal/common"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/exporter/otlphttpexporter"
+)
+
+func init() {
+	converters = append(converters, otlpHTTPExporterConverter{})
+}
+
+type otlpHTTPExporterConverter struct{}
+
+func (otlpHTTPExporterConverter) Factory() component.Factory {
+	return otlphttpexporter.NewFactory()
+}
+
+func (otlpHTTPExporterConverter) InputComponentName() string {
+	return "otelcol.exporter.otlphttp"
+}
+
+func (otlpHTTPExporterConverter) ConvertAndAppend(state *state, id component.InstanceID, cfg component.Config) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	label := state.FlowComponentLabel()
+
+	args := toOtelcolExporterOTLPHTTP(cfg.(*otlphttpexporter.Config))
+	block := common.NewBlockWithOverride([]string{"otelcol", "exporter", "otlphttp"}, label, args)
+
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Converted %s into %s", stringifyInstanceID(id), stringifyBlock(block)),
+	)
+
+	state.Body().AppendBlock(block)
+	return diags
+}
+
+func toOtelcolExporterOTLPHTTP(cfg *otlphttpexporter.Config) *otlphttp.Arguments {
+	return &otlphttp.Arguments{
+		Client:       otlphttp.HTTPClientArguments(toHTTPClientArguments(cfg.HTTPClientSettings)),
+		Queue:        toQueueArguments(cfg.QueueSettings),
+		Retry:        toRetryArguments(cfg.RetrySettings),
+		DebugMetrics: common.DefaultValue[otlphttp.Arguments]().DebugMetrics,
+	}
+}
+
+func toHTTPClientArguments(cfg confighttp.HTTPClientSettings) otelcol.HTTPClientArguments {
+	var mic *int
+	var ict *time.Duration
+	defaults := confighttp.NewDefaultHTTPClientSettings()
+	if mic = cfg.MaxIdleConns; mic == nil {
+		mic = defaults.MaxIdleConns
+	}
+	if ict = cfg.IdleConnTimeout; ict == nil {
+		ict = defaults.IdleConnTimeout
+	}
```

Comment:
This is because these [two defaults](https://github.com/open-telemetry/opentelemetry-collector/blob/ae90430559597f83193258ec8a2789626d6d23b0/config/confighttp/confighttp.go#L121-L126) are only set as pointers in the NewDefaultClientConfig constructor of the confighttp package.

If you spot of any better way to do this, let me know.

Reply:
Nah I think this is a good way of doing it.